osmoutils/accum/accum_test.go

```diff
@@ -1461,3 +1462,95 @@ func (suite *AccumTestSuite) TestSetPositionCustomAcc() {
 		})
 	}
 }
+
+// We run a series of partially random operations on two accumulators to ensure that total shares are properly tracked in state
+func (suite *AccumTestSuite) TestGetTotalShares() {
+	suite.SetupTest()
+
+	// Set up two accumulators to ensure we can test all relevant behavior
+	err := accumPackage.MakeAccumulator(suite.store, testNameOne)
+	suite.Require().NoError(err)
+	err = accumPackage.MakeAccumulator(suite.store, testNameTwo)
+	suite.Require().NoError(err)
+
+	// Fetch both accumulators for initial sanity tests
+	accumOne, err := accumPackage.GetAccumulator(suite.store, testNameOne)
+	suite.Require().NoError(err)
+	accumTwo, err := accumPackage.GetAccumulator(suite.store, testNameTwo)
+	suite.Require().NoError(err)
+
+	// Ensure that both accums start at zero shares
+	suite.Require().Equal(sdk.ZeroDec(), accumOne.GetTotalShares())
+	suite.Require().Equal(sdk.ZeroDec(), accumTwo.GetTotalShares())
+
+	// Create position on first accum and pull new accum objects from state
+	err = accumOne.NewPosition(testAddressOne, sdk.OneDec(), nil)
+	suite.Require().NoError(err)
+	accumOne, err = accumPackage.GetAccumulator(suite.store, testNameOne)
+	suite.Require().NoError(err)
+	accumTwo, err = accumPackage.GetAccumulator(suite.store, testNameTwo)
+	suite.Require().NoError(err)
+
+	// Check that total shares for accum one has updated properly and accum two shares are unchanged
+	suite.Require().Equal(sdk.OneDec(), accumOne.GetTotalShares())
+	suite.Require().Equal(sdk.ZeroDec(), accumTwo.GetTotalShares())
+
+	// Run a number of NewPosition, AddToPosition, and RemoveFromPosition operations on each accum
+	testAddresses := []string{testAddressOne, testAddressTwo, testAddressThree}
+	accums := []accumPackage.AccumulatorObject{accumOne, accumTwo}
+	expectedShares := []sdk.Dec{sdk.OneDec(), sdk.ZeroDec()}
+
+	for i := 1; i <= 10; i++ {
+		// Cycle through accounts and accumulators
+		curAddr := testAddresses[i % 3]
+		curAccum := accums[i % 2]
+
+		// We set a baseAmt that varies with the iteration to increase coverage
+		baseAmt := sdk.NewDec(int64(i)).Mul(sdk.NewDec(10))
+
+		// If addr doesn't have a position yet, we make one
+		positionExists, err := curAccum.HasPosition(curAddr)
+		suite.Require().NoError(err)
+		if !positionExists {
+			err = curAccum.NewPosition(curAddr, baseAmt, nil)
+			suite.Require().NoError(err)
+		}
+
+		// We generate a random binary value (0 or 1) to determine
+		// whether we will add and/or remove liquidity this loop
+		addShares := sdk.NewDec(int64(rand.Int()) % 2)
+		removeShares := sdk.NewDec(int64(rand.Int()) % 2)
```

Comment:
We should be setting a deterministic seed for the randomness. Otherwise, it would be difficult to debug non-determinis if any ever occurs

Reply:
Good call!